kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiServer.scala

```diff
@@ -105,6 +105,10 @@ object KyuubiServer extends Logging {
     val _hadoopConf = KyuubiHadoopUtils.newHadoopConf(new KyuubiConf().loadFileDefaults())
     hadoopConf = _hadoopConf
   }
+
+  private[kyuubi] def reloadServerConf(): Unit = synchronized {
+    kyuubiServer.conf.loadFileDefaults()
```

Comment:
@turboFei @pan3793 
Have a look at this PR if you have time.
It's reusing the `loadFileDefaults` method to read the config file  `kyuubi-defaults.conf` and overriding server configs. And if any configs not in the config file are set during runtime, they won't be affected.